### DISCUSSION TITLE
Contig assembly question (MiXCR vs. Cellranger)

### DISCUSSION THREAD

**Discussion by larryyang1980** [milaboratory/mixcr#1294] [Q&A] [answered]
Hi, has anyone ever compared the contig assembly outputs between MiXCR and Cellranger? I did it for the same dataset based on the same IMGT reference. I noticed there were a lot of contigs from Cellranger only, which didn't appear in the MiXCR outputs. I'm still trying to figure out which tool is better for my study. Any advice or suggestion would be great! Thanks!

![MiXCRvsCellranger_TRA](https://github.com/milaboratory/mixcr/assets/7346479/a3ff0368-512b-4c18-a64e-86f0b53c7893)
![MiXCRvsCellranger_TRB](https://github.com/milaboratory/mixcr/assets/7346479/a7764c63-3e72-41cb-8aea-dbcaa33b0457)

**Comment by Alex-Davydov** [comment]
Hi! What version of MiXCR and exact command was used for analysing this sample?

**Reply by larryyang1980** [reply]
MiXCR v4.3.2 (built Tue Apr 11 14:35:07 EDT 2023; rev=123d699964; branch=no_branch; host=fv-az364-276)
RepSeq.IO v1.8.0-21-master (rev=5521c5842b)
MiLib v2.4.0-7-master (rev=acea8397fe)
Built-in V/D/J/C library: repseqio.v2.2

```shell
mixcr -Xmx50g analyze -s mmu --library imgt --force-overwrite 10x-vdj-tcr TCR_R1.fastq.gz TCR_R2.fastq.gz TCR_imgt_lib
``` 

Cellranger version 7.1.0
```shell
cellranger vdj --id=TCR_Cellranger --reference=/files/vdj_ref_IMGT --fastqs=/files/runs/HF2Y5DRX2/outs/fastq_path --sample=TCR --localcores=8
```

**Comment by Alex-Davydov** [comment]
Hi, we suggest using the newest version of MiXCR (4.4.2) for extracting TCR/BCR chains, as improvements have been made in 4.4.x versions. The preset name has also been updated to `10x-sc-xcr-vdj`, and it's now compatible with all B/T cell chains. When comparing it with cellranger, the difference is typically no more than 5%. This discrepancy occurs because the filtering algorithms are not identical, making it impossible to achieve the exact same result. You can execute the following command for the latest version:

`mixcr analyze 10x-sc-xcr-vdj -s mmu --library IMGT --force-overwrite TCR_R1.fastq.gz TCR_R2.fastq.gz TCR_imgt_lib/`

**Comment by mizraelson** [comment, marked as answer]
As mentioned earlier, the results between MiXCR and CellRanger might differ due to different filtering algorithms. Additionally, the differences may change from sample to sample. I have analyzed 62 samples using the latest MiXCR (v.4.4.2) and CellRanger (v7.1.0) versions, and you can find the comparison in the datatable attached.

The plots below show that in most cases, the number of identified TRA and TRB chains is similar between the two platforms.
<img width="620" alt="image" src="https://github.com/milaboratory/mixcr/assets/18702359/30b93905-b4e2-4c9a-b4be-2914c06c6b6c">
<img width="589" alt="image" src="https://github.com/milaboratory/mixcr/assets/18702359/1db14972-b0b6-46fb-9cff-5fdf47921716">


If we calculate t-statistics and p-values to demonstrate whether MiXCR identifies a significantly different number of chains compared to CellRanger, we get the following numbers:

    TRB t-statistic: 3.7345319524069613, p-value: 0.00041590120277170285
    TRA t-statistic: 3.5012084198174476, p-value: 0.0008723899105857749

A slightly positive t-statistic suggests that, on average, MiXCR is identifying more chains than CellRanger for both TRB and TRA.

Below, you can also find the comparison of the number of cells for which both chains were identified by the two platforms.
Hi, as mentioned earlier the results between MiXCR and CellRanger might differ due to different filtering algorithms. Also, the difference changes from sample to sample. Below you can see a comparison of 62 samples analyzed with the latest MiXCR (v.4.4.2) and Cell Ranger (v7.1.0) versions. Find the datatable attached.





The plots show that in most cases the number of identified TRA and TRB chains is similar between two platforms.

If we calculate t-statistics and p-values to demonstrate whether MiXCR identifies a significantly different number of chains compared to Cell Ranger we get the following numbers:

TRB t-statistic: 3.7345319524069613, p-value: 0.00041590120277170285 
TRA t-statistic: 3.5012084198174476, p-value: 0.0008723899105857749

A slightly positive t-statistic suggests that, on average, MiXCR is identifying more chains than Cell Ranger for both TRB and TRA.


Below you can also find the comparison of the number of cells for which both chains were identified by two platforms.
<img width="530" alt="image" src="https://github.com/milaboratory/mixcr/assets/18702359/21a1f194-5a9a-4a45-99a2-9b2406964449">


[mixcrVScellranger.txt](https://github.com/milaboratory/mixcr/files/12352567/mixcrVScellranger.txt)
